### PR TITLE
Fix support for the upcoming Neon release

### PR DESCRIPTION
**Pull request by s0undt3ch**
### What does this PR do?
See title.
Additionally:
* switch fr drone to github actions.
* Add pre commit support on the repo

### CODE REVIEW

**Review by vutny — COMMENTED** [review]
Finally, Drone has gone! :smile_cat: 

I guess the file `Dockerfile.drone-builds` and the bage and reference to Drone build in `README` is no longer needed, right?

**Comment by s0undt3ch** [comment]
Unfortunately this was kind of wishful thinking...
If we have a paid plan with Github, and it's an old plan, even though the project is public, we don't get access to github actions, which doesn't make sense...
SaltStack has a paid plan, and it's not one of the new plans, so, no github actions on this or any publicly available repos under the SaltStack ORG, at least for the short term.

**Comment by vutny** [comment]
:crying_cat_face:

**Comment by s0undt3ch** [comment]
Indeed.

**Comment by s0undt3ch** [comment]
You can see it work [here](https://github.com/s0undt3ch/salt-bootstrap/actions) though... just for the (sad) fun of it.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v2.1.0
+    hooks:
+      - id: check-merge-conflict  # Check for files that contain merge conflict strings.
+      - id: trailing-whitespace   # Trims trailing whitespace.
+        args: [--markdown-linebreak-ext=md]
+      - id: mixed-line-ending     # Replaces or checks mixed line ending.
+        args: [--fix=lf]
+      - id: end-of-file-fixer     # Makes sure files end in a newline and only a newline.
+      - id: check-merge-conflict  # Check for files that contain merge conflict strings.
```
Comment (waynew) [inline .pre-commit-config.yaml:13]: duplicate `check-merge-conflict` hooks, unless that's intended?

**Review by waynew — DISMISSED** [review]

File: .pre-commit-config.yaml (hunk 1 above)
Comment (s0undt3ch) [reply .pre-commit-config.yaml:13]: Yes, duplicated. Bah!
I'll fix it in another PR since this one is already passing

Comment (waynew) [reply .pre-commit-config.yaml:13]: Do the checks really take that long in this repo?

Comment (s0undt3ch) [reply .pre-commit-config.yaml:13]: Depends on the drone queue.

**Pull request merged**